config.go

```diff
@@ -1355,6 +1360,16 @@ func ValidateConfig(cfg Config, usageMessage string,
 		return nil, err
 	}
 
+	// Log a warning if our expiry delta is not greater than our incoming
+	// broadcast delta. We do not fail here because this value may be set
+	// to zero to intentionally keep lnd's behavior unchanged from when we
+	// didn't auto-cancel these invoices.
+	if cfg.Invoices.HoldExpiryDelta <= lncfg.DefaultIncomingBroadcastDelta {
+		ltndLog.Warnf("Invoice hold expiry delta: %v <= incoming " +
```

Comment:
Auto-expire does change some assumptions around hold invoices. 
- Previously: You have until the htlc's expiry to go to chain, even if that means a channel is force closed
- Now: The invoice will be canceled before it goes to chain, so you won't be able to settle after `expiry - HoldExpiryDelta`

I thought that it would be good to allow the option to opt-out of this behaviour change by allowing a 0 value delta. This still lets you settle up until `expiry`, and the invoice registry state will be handled properly because we will cancel any un-settled invoices at their expiry height even after they've gone to chain. 

For 99.99% of people I don't think this value will be updated, and for the rest it's nice to have the option? We could come back to this later once it's been well communicated and enforce? Since there isn't really much use-case for allowing your channels to be force closed IMO 😅

Reply:
Fair enough. The warning is probably scary enough to make users think twice. Maybe worth thinking through why a 0 delta would add to the overall (power)user experience.

Reply:
It's more about giving the option to not change behaviour. Eg, I was thinking about loop, with a 12 block delta, we won't be able to settle our invoices 12 blocks before expiry, because they will be auto-cancelled which changes assumptions a bit. I think it's nice to give services like loop the option to disable this change while they update to accommodate it, then we can enforce it in the next major release?